Sources/Purchasing/StoreKit2/StoreKit2TransactionListener.swift

```diff
@@ -96,19 +100,18 @@ private extension StoreKit2TransactionListener {
             )
 
         case .verified(let verifiedTransaction):
-            let customerInfo = try await self.finish(transaction: verifiedTransaction)
+            await self.finish(transaction: verifiedTransaction)
 
-            return (verifiedTransaction, customerInfo)
+            if notifyDelegate, let delegate = self.delegate {
+                _ = try await delegate.transactionsUpdated()
+            }
```

Comment:
This avoids calling the delegate during purchases (leading to a duplicate `syncPurchases` call). Now this is only called for transactions that come from `StoreKit.Transaction.updates`